src/plugins/data/server/search/opensearch_search/decide_client.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { OpenSearchClient, RequestHandlerContext } from 'src/core/server';
+import { IOpenSearchSearchRequest } from '..';
+
+export const decideClient = async (
```

Comment:
maybe `decideClient ` -> `clientStrategy` ?

Reply:
function is an action of behavior, I'd prefer to keep the existing name that conveys an idea of action